pkg/cvo/availableupdates.go

```diff
@@ -67,14 +67,18 @@ func (optr *Operator) syncAvailableUpdates(ctx context.Context, config *configv1
 	}
 
 	userAgent := optr.getUserAgent()
-
-	current, updates, conditionalUpdates, condition := calculateAvailableUpdatesStatus(ctx, string(config.Spec.ClusterID),
+	clusterId := string(config.Spec.ClusterID)
+	current, updates, conditionalUpdates, condition := calculateAvailableUpdatesStatus(ctx, clusterId,
 		transport, userAgent, upstream, desiredArch, currentArch, channel, optr.release.Version, optr.conditionRegistry)
 
 	if usedDefaultUpstream {
 		upstream = ""
 	}
 
+	if optr.isHCPModeEnabled {
```

Comment:
> we may want to test for misconfigured empty clusterID?

Is the question regarding whether the substitution of `_id` works for empty `clusterId`? Or to see whether something breaks and potential side effects?

> Also, are the hosted custer admins able to spoof their clusters ClusterID and read other hosted cluster metrics this way? If yes then that seems to be somewhat security-sensitive...

Good question. The clusterId and similar things in the cluster version are being reconciled from the HCP (hosted control plane) and should be overwritten.

https://github.com/openshift/hypershift/blob/main/control-plane-operator/hostedclusterconfigoperator/controllers/resources/resources.go#L956-L960

```
func (r *reconciler) reconcileClusterVersion(ctx context.Context, hcp *hyperv1.HostedControlPlane) error {
	clusterVersion := &configv1.ClusterVersion{ObjectMeta: metav1.ObjectMeta{Name: "version"}}
	if _, err := r.CreateOrUpdate(ctx, r.client, clusterVersion, func() error {
		clusterVersion.Spec.ClusterID = configv1.ClusterID(hcp.Spec.ClusterID)
		clusterVersion.Spec.Capabilities = nil
		clusterVersion.Spec.Upstream = ""
		clusterVersion.Spec.Channel = hcp.Spec.Channel
		clusterVersion.Spec.DesiredUpdate = nil
		return nil
	}); err != nil {
		return fmt.Errorf("failed to reconcile clusterVersion: %w", err)
	}

	return nil
}
```

But I am not sure which roles are binded to the hosted cluster admins at the moment. The `admin-kubeconfig` secret that is available in the HCP does provide the capability to change the cluster version. Although I need to check whether these kind of permissions are binded to a normal hosted cluster admin...

Reply:
The question I think Petr was asking if the clusterID is empty for some reason(because of a bug in the code) do you feel confident that it is handled properly with right information in the log.